src/utils/logic.ts

```diff
@@ -50,25 +62,28 @@ const evalComparisonRule = (
   rule: ResolvedComparisonRule,
   fieldValues: { [key: string]: any }
 ): boolean => {
-  const fieldValue = fieldValues[rule.field_key];
-  return (
-    Array.isArray(fieldValue)
-      ? fieldValue.length
-        ? fieldValue
-        : [undefined]
-      : [fieldValue]
-  )
+  // flatten the right side values/fields into flat list of values
+  const flatValues = rule.values.flatMap((value) => {
+    if (value !== null && valueTypeIsField(value))
```

Comment:
if (value?.field_key)

maybe woudl be simpler?

Reply:
Unfortunately, no.  Typescript needs narrowing here and the best way is the type predicate with the typeof guard.